pypi-vuln/index-2017-11-08-pypirc_exposure_on_github.rst

```diff
@@ -0,0 +1,109 @@
+identify 77 valid PyPI logins in 85 public files published to GitHub. These 77
+logins had maintainer or administrator access to 146 unique projects on PyPI.
+
+Audit
+-----
+Action Taken by PyPI team
+
+The PyPI security team followed up by auditing and extending the Proof of
+Concept tools supplied by the reporter to verify the report.
+
+After running the tooling against the full result set of the GitHub code search
+the PyPI administrators unset the passphrases for all valid logins found and
+issued an administrative password reset for exposed users.
+
+Additionally an audit of PyPI's journals showed no signs of malicious access
+for the exposed accounts.
+
+The email sent to affected users took the form
+
+.. code-block::
+
+  From: admin@mail.pypi.python.org
+  To: {user['email']}
+  Subject: [Urgent] Your PyPI password has been reset
+  
+  {username},
+  
+  A security report recently identified that your PyPI login credentials were
+  exposed in a public code repository on github.com.
+  
+  Please see the following links where your credentials were found:
+  
+  {pypirc_links}
+  
+  An initial audit of our journals found that {package_count} projects your
+  account has access to were potentially exposed but did not indicate any
+  malicious activity.
+  
+  Packages:
+  
+  {packages}
+  
+  Please double check the audit logs at https://pypi.python.org after you have
+  reset your password and notify us if you identify any suspicious activity.
+  
+  Also please reset your passwords anywhere else you may have used the password
+  exposed in the above links.
+  
+  To reset your password, please visit {password_reset_link}.
+  
+  Thanks,
+  PyPI Security Team
+
+Recommendations
+---------------
+
+All users of PyPI should ensure that their PyPI login credentials are safe and
+have not been inadvertently exposed in a public repository of dotfiles, in the
+root of a project directory, or in some other public or shared medium.
+
+The PyPI team does not have the resources to search or scrape all such services
+and may not have identified all forms of this exposure.
+
+Additionally, reviewing the Audit Journal for your projects on pypi.python.org
+for suspicous activity is a good idea. If you identify any such activity,
+please report it per `our published security policy <https://pypi.org/security/>`_.
+
+Timeline
+--------
+
+Timeline using the disclosure date **2017-11-05** as reference:
+
+* 2017-11-05 Issue reported by Joachim Jablon to a single member of the security team listed in our security policy on `pypi.org <https://pypi.org/security/>`_
+* 2017-11-08 (**+3days**):Issue reported by Joachim Jablon to an additional member of the security team listed in our security policy on `pypi.org <https://pypi.org/security/>`_
+* 2017-11-08 (**+3days**):Issue reported by Joachim Jablon to all members of the security team listed in our security policy on `pypi.org <https://pypi.org/security/>`_
```

Comment:
@ewdurbin So about that: Reading this, it feels like I messed up, and I admit I did, but importing all the public keys didn't work for some reason, and the security page of pypi.org says "please email Donald Stufft, Ernest W. Durbin III, **or** Richard Jones". Would you mind creating a single alias (pypi-security@python.org or security@ pypi.org or something) with a single gpg key ?